source/assets/scripts/backend.js

```diff
@@ -0,0 +1,344 @@
+/** @module backend */
+
+import { recipe_data, keep_fields } from './recipe-data.js';
+
+const API_KEY = '486eca841c6a49b896486723439f9977';
+const CUSTOM_RECIPE_KEY = '%custom_recipes';
+const FAVORITE_RECIPE_KEY = '%favorite_recipes';
+const INTOLERANCE_KEY = '%intolerances';
+
+/**
+ * search a recipe by its name and return a promise of list of raw json
+ * @param {string} name - name of the recipe
+ * @param {Array<string>} intolerances - a list of intolerances. see below for details
+ * @returns {Promise} - a list of unfiltered recipe, empty if non found
+ */
+async function fetch_recipe_raw(name, intolerances) {
+  const intolerances_str = Array.isArray(intolerances) ? `&intolerances=${intolerances.join(',')}` : '';
+  const url = `https://api.spoonacular.com/recipes/complexSearch?query=${name}${intolerances_str}&apiKey=${API_KEY}&addRecipeInformation=true&fillIngredients=true`;
+  const response = await fetch(url);
+  const data = await response.json();
+  if (data.results) return data.results;
+  else return [];  // an empty list
+}
+
+/**
+ * search a recipe by its name and keep only the parts we need according to the global variable keep_fields
+ * we then remap the field names by the rule fields_remap
+ * afterwards, it saves all filtered recipes to the localstore
+ * @param {string} name - name of the recipe 
+ * @param {Array<string>} intolerances - a list of intolerances. for a list of supported keywords, see 
+ *  https://spoonacular.com/food-api/docs#:~:text=of%20supported%20diets.-,intolerances,-string
+ *  Note that this functionality is pretty bad since if a recipe can be nut free without listing nuts as intolerance
+ * @return {Promise} returns a list of fetched recipe for testing purpose
+ */
+export async function fetch_recipe(name, intolerances) {
```

Comment:
I think we are going to ignore the intolerances and do our own filtering by the ingredients